bindings/go/runtime/json_schema_generator.go

```diff
@@ -0,0 +1,46 @@
+package runtime
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/invopop/jsonschema"
+)
+
+// GenerateJSONSchemaForType takes a Type and uses reflection to generate a JSON Schema representation for it.
+// It will also use the correct type representation as we don't marshal the type in object format.
+func GenerateJSONSchemaForType(obj Typed) ([]byte, error) {
+	if obj == nil {
+		return nil, fmt.Errorf("cannot generate JSON schema for nil object")
+	}
+
+	switch v := obj.(type) {
+	case *Unstructured:
+		if v.Data == nil {
```

Comment:
can we generate a json scheme for a nil object? that would be weird wouldnt it?

Reply:
No, I don't think it can generate anything from a nil object indeed. That's why we error out here if Data is nil.

Reply:
if we cant generate anything from a nil then why dont we just return an error no matter if we are nil or not?

Reply:
Err, I didn't quite get that, sorry. We can generate something from something though. So if it's not nil, it will create something like :

```json
...{"Unstructured":{"properties":{"Data":{"type":"object"}}...
```

Is that not good?

Reply:
no because theres no property such as Data :)

Reply:
you mean no explicit property, it's the thing _inside_ data that you want in the schema

Reply:
no i mean that Data is not a valid property of that object. its an incorrect schema.

Reply:
Okay, so basically, just error out for now if the type is either Unstructured or Raw. :)